packages-exp/auth-exp/src/core/util/assert.ts

```diff
@@ -15,28 +15,72 @@
  * limitations under the License.
  */
 
+import { SDK_VERSION } from '@firebase/app-exp';
 import { AUTH_ERROR_FACTORY, AuthErrorCode } from '../errors';
+import { logError } from './log';
 
-export function assert<T>(
-  expression: T | null | undefined,
-  appName: string
-): T {
-  if (!expression) {
-    throw AUTH_ERROR_FACTORY.create(AuthErrorCode.INTERNAL_ERROR, { appName });
-  }
+/**
+ * Unconditionally fails, throwing a developer facing INTERNAL_ERROR
+ *
+ * @param appName App name for tagging the error
+ * @throws FirebaseError
+ */
+export function fail(appName: string): never {
+  throw AUTH_ERROR_FACTORY.create(AuthErrorCode.INTERNAL_ERROR, { appName });
+}
 
-  return expression;
+/**
+ * Verifies the given condition and fails if false, throwing a developer facing error
+ *
+ * @param assertion
+ * @param appName
+ */
+export function assert(assertion: boolean, appName: string): asserts assertion {
+  if (!assertion) {
+    fail(appName);
+  }
 }
 
-export function assertType<T>(
-  expression: unknown,
-  expected: string | string[],
+export function assertStringOrUndefined(
+  assertion: unknown,
   appName: string
-): T {
-  if (typeof expected === 'string') {
-    expected = [expected];
-  }
+): asserts assertion is string | undefined {
+  assert(
+    typeof assertion === 'string' || typeof assertion === 'undefined',
+    appName
+  );
+}
+
+/**
+ * Unconditionally fails, throwing an internal error with the given message.
+ *
+ * @param failure type of failure encountered
+ * @throws Error
+ */
+export function debugFail(failure: string): never {
+  // Log the failure in addition to throw an exception, just in case the
+  // exception is swallowed.
+  const message = `AUTH (${SDK_VERSION}) INTERNAL ASSERTION FAILED: ` + failure;
```

Comment:
Looks like the logError function already includes Auth and the sdk version. Can probably remove those bits